pkg/operator/service.go

```diff
@@ -99,3 +103,21 @@ func (op *Operator) watchService() {
 	)
 	ctrl.Run(wait.NeverStop)
 }
+
+func updateServiceMonitorIndex(promClient pcm.MonitoringV1alpha1Interface, idx indexers.ServiceMonitorIndexer) error {
+	// Check if ServiceMonitor Indexer is activated
+	if idx == nil {
+		return nil
+	}
+
+	serviceMonitors, err := promClient.ServiceMonitors(apiv1.NamespaceAll).List(metav1.ListOptions{})
```

Comment:
I see. Move this method inside the SMIndexer (same for SvcIndexer).

Also, add a `Add(svcMon, svc)` type method to avoid n^2 loops.